### ISSUE TITLE
Postgres error when sending concurrent webhooks

### ISSUE THREAD

**Issue by JordanSussman** [go-vela/community#213]
## Description

Currently, sending webhook payloads for the same repo in quick succession can cause the error of `pq: duplicate key value violates unique constraint "builds_repo_id_number_key"`.

## Value

Vela is able to receive near concurrent webhooks without failing.

## Useful Information

Steps to reproduce:

1. Activate repo in Vela
2. Send webhook payload in quick succession (e.g. `curl -H "Content-Type: application/json" -H "X-GitHub-Event: push" -H "X-GitHub-Delivery: <id>" -X POST http://localhost:8080/webhook -d "$(cat payload.json)"`)

You can also sporadically reproduce the error if you attempt to quickly restart builds within the UI.

1. What is the output of `vela --version`?

Server is utilizing 0.7.2

2. What operating system is being used?

MacOS

**Comment by jbrockopp** [comment]
Related PRs that attempted to resolve this issue previously:

* https://github.com/go-vela/server/pull/186
* https://github.com/go-vela/server/pull/191

**Comment by wass3r** [comment]
Is this still an issue with `0.7.x`?

**Comment by JordanSussman** [comment]
> Is this still an issue with `0.7.x`?

Yes - this issue became increasingly apparent when I was spamming builds to reproduce https://github.com/go-vela/community/issues/83.

**Comment by wass3rw3rk** [comment]
should be solved in v28
